tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/generator/ResourceFallbackCodeGenerator.java

```diff
@@ -0,0 +1,181 @@
+package main.java.org.unicode.icu.tool.cldrtoicu.generator;
+
+import com.google.common.base.Splitter;
+import main.java.org.unicode.icu.tool.cldrtoicu.CodeGenerator;
+import org.unicode.cldr.api.*;
+
+import java.io.PrintWriter;
+import java.nio.file.Path;
+import java.util.*;
+
+import static com.google.common.base.CharMatcher.whitespace;
+
+public class ResourceFallbackCodeGenerator implements CodeGenerator {
+    private Map<String, String> defaultScripts;
+    private Map<String, String> parentLocales;
+    private Splitter localeIDSplitter;
+    private Splitter childLocaleSplitter;
+
+    @Override
+    public void generateCode(Path cldrPath, PrintWriter cFileOut, PrintWriter javaFileOut) {
+        defaultScripts = new TreeMap<String, String>();
+        parentLocales = new TreeMap<String, String>();
+        localeIDSplitter = Splitter.on('_');
+        childLocaleSplitter = Splitter.on(whitespace()).omitEmptyStrings();
+
+        CldrDataSupplier supplier = CldrDataSupplier.forCldrFilesIn(cldrPath);
+        CldrData supplementalData = supplier.getDataForType(CldrDataType.SUPPLEMENTAL);
+        supplementalData.accept(CldrData.PathOrder.NESTED_GROUPING, new CldrData.PrefixVisitor() {
+            @Override
+            public void visitPrefixStart(CldrPath prefix, Context context) {
+                if (prefix.getName().endsWith("likelySubtags")) {
+                    context.install(cldrValue -> handleLikelySubtag(cldrValue));
+                } else if (prefix.getName().endsWith("parentLocales")) {
+                    context.install(cldrValue -> handleParentLocale(cldrValue));
+                }
+            }
+        });
+
+        generateCFile(cFileOut);
+        generateJavaFile(javaFileOut);
+    }
+
+    private void handleLikelySubtag(CldrValue value) {
+        String from = value.get(AttributeKey.keyOf("likelySubtag", "from"));
+        String to = value.get(AttributeKey.keyOf("likelySubtag", "to"));
+
+        String[] fromPieces = localeIDSplitter.splitToList(from).toArray(new String[] {});
+        String[] toPieces = localeIDSplitter.splitToList(to).toArray(new String[] {});
+
+        if (toPieces.length != 3) {
+            System.err.println("Didn't get 3 segments in 'to' value: from=" + from + ", to=" + to);
+            return;
```

Comment:
Why not throw an exception? What if someone runs the converter and misses the console output?

Reply:
I don't know.  I think that felt too drastic, and I wasn't sure when I wrote this that I was guaranteed that the likely-subtags resources would all have three-segment locale IDs (yeah, I know that's the whole point, but I was worried there might be exceptions to it).  In other words, it was as much about my confirming my own assumptions as anything.

If I change this to throw an exception, what exception should I throw?  And would I also do it on line 59 (the "'from' entry has a non-'und' language..." thing)?

Reply:
How about IllegalArgumentException in both places? I am not particular about which exception to throw here.

Reply:
Okay.